src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs

```diff
@@ -254,31 +265,66 @@ private async Task CheckLoginRequestStatus()
             }
         }
 
+        private async Task HandleLoginComplete()
+        {
+            await _stateService.SetPendingAdminAuthRequestAsync(null);
+            _syncService.FullSyncAsync(true).FireAndForget();
+            LogInSuccessAction?.Invoke();
+        }
+
         private async Task CreatePasswordlessLoginAsync()
         {
             await Device.InvokeOnMainThreadAsync(() => _deviceActionService.ShowLoadingAsync(AppResources.Loading));
 
-            var response = await _authService.PasswordlessCreateLoginRequestAsync(_email, AuthRequestType);
-            if (response != null)
+            PasswordlessLoginResponse response = null;
+            var pendingRequest = await _stateService.GetPendingAdminAuthRequestAsync();
+            if (pendingRequest != null && _authRequestType == AuthRequestType.AdminApproval)
```

Comment:
Do we need to build the flow for sending a regular device approval here?

Reply:
I was under the impression that only _**Admin Approval Request**_ had the pending feature.

Reply:
Confirmed on `clients` repo and only `AuthRequestType.AdminApproval` has pending feature.

Reply:
it's handling both, it first checks for pending admin requests. After that, if the response is still null it creates a new request based on Request Type:

`response = await _authService.PasswordlessCreateLoginRequestAsync(_email, **AuthRequestType**);`